frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/hooks/k8s-watcher.ts

```diff
@@ -47,15 +47,25 @@ export const makeQuery: MakeQuery = (namespace, labelSelector, fieldSelector, na
   return query;
 };
 
+const CACHE_SYMBOL = Symbol('_cachedToJSResult');
+
 export const getReduxData = (immutableData, resource: WatchK8sResource) => {
   if (!immutableData) {
     return null;
   }
   if (resource.isList) {
-    return immutableData.toArray().map((a) => a.toJSON());
+    return immutableData.toArray().map((a) => {
+      if (!a[CACHE_SYMBOL]) {
+        a[CACHE_SYMBOL] = a.toJSON();
+      }
+      return a[CACHE_SYMBOL];
+    });
   }
-  if (immutableData.toJSON) {
-    return immutableData.toJSON();
+  if (immutableData.toJS) {
```

Comment:
Shouldn't this still be:
```suggestion
  if (immutableData.toJSON) {
```